contracts/cw-escrow/src/contract.rs

```diff
@@ -0,0 +1,225 @@
+#[cfg(not(feature = "library"))]
+use cosmwasm_std::entry_point;
+use cosmwasm_std::{to_binary, Addr, Binary, Deps, DepsMut, Env, MessageInfo, Response, StdResult};
+use cw2::set_contract_version;
+use cw_utils::must_pay;
+
+use crate::{
+    error::ContractError,
+    msg::{ExecuteMsg, InstantiateMsg, QueryMsg, StatusResponse},
+    state::{CheckedTokenInfo, COUNTERPARTY_ONE, COUNTERPARTY_TWO},
+};
+
+const CONTRACT_NAME: &str = "crates.io:cw-escrow";
+const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn instantiate(
+    deps: DepsMut,
+    _env: Env,
+    _info: MessageInfo,
+    msg: InstantiateMsg,
+) -> Result<Response, ContractError> {
+    set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
+
+    let counterparty_one = msg.counterparty_one.into_checked(deps.as_ref())?;
+    let counterparty_two = msg.counterparty_two.into_checked(deps.as_ref())?;
+
+    COUNTERPARTY_ONE.save(deps.storage, &counterparty_one)?;
+    COUNTERPARTY_TWO.save(deps.storage, &counterparty_two)?;
+
+    Ok(Response::new()
+        .add_attribute("method", "instantiate")
+        .add_attribute("counterparty_one", counterparty_one.address)
+        .add_attribute("counterparty_two", counterparty_two.address))
+}
+
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn execute(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+    msg: ExecuteMsg,
+) -> Result<Response, ContractError> {
+    match msg {
+        ExecuteMsg::Receive(msg) => execute_receive(deps, info.sender, msg),
+        ExecuteMsg::Fund {} => execute_fund(deps, info),
+        ExecuteMsg::Withdraw {} => execute_withdraw(deps, info),
+    }
+}
+
+pub fn execute_receive(
```

Comment:
Feels like a lot of duplicated code, can we make execute receive funnel into execute fund?